pkg/asset/installconfig/gcp/validation.go

```diff
@@ -876,3 +877,25 @@ func validateServiceEndpointOverride(client API, ic *types.InstallConfig, fieldP
 
 	return allErrs
 }
+
+// validateFirewallPermissions validates that the user has firewall permissions OR when the user does not have
+// permissions to create firewall rules then a network is specified.
+func validateFirewallPermissions(ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	projectID := ic.GCP.ProjectID
+	configField := "projectID"
+	if ic.GCP.NetworkProjectID != "" {
+		projectID = ic.GCP.NetworkProjectID
+		configField = "networkProjectID"
+	}
+
+	createFwRules, err := HasPermissions(context.TODO(), projectID, []string{CreateGCPFirewallPermission}, ic.GCP.Endpoint)
+	if err != nil {
+		return append(allErrs, field.Invalid(fieldPath.Child(configField), ic.GCP.NetworkProjectID, err.Error()))
+	}
+	if !createFwRules && ic.GCP.Network == "" {
+		allErrs = append(allErrs, field.Required(fieldPath.Child("network"), "fw rule creation permission is missing, an existing network is required"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Required(fieldPath.Child("network"), "firewall rule creation permission is missing, an existing network is required"))
```

nit: Just clearer for users :grin: